src/content/docs/en/recipes/dynamically-importing-images.mdx

```diff
@@ -0,0 +1,146 @@
+        altText: string;
+        name: string;
+        age: number;
+     }
+
+     const { imagePath, altText, name, age } = Astro.props;
+    ---
+    ``` 
+
+4. Create a new `images` variable and use the `import.meta.glob` function which returns an object of all of the image paths inside the `assets` folder. 
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={12}
+    ---
+     import { Image } from 'astro:assets';
+
+     interface Props {
+        imagePath: string;
+        altText: string;
+        name: string;
+        age: number;
+     }
+     
+     const { imagePath, altText, name, age } = Astro.props;
+     const images = import.meta.glob('/src/**/*.{jpeg,jpg,png,gif}')
+    ---
+    ```
+
+5. Use the props to create the markup for your card component. 
+
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={14-18}
+    ---
+     import { Image } from 'astro:assets';
+
+     interface Props {
+        imagePath: string;
+        altText: string;
+        name: string;
+        age: number;
+     }
+     
+     const { imagePath, altText, name, age } = Astro.props;
+     const images = import.meta.glob('/src/**/*.{jpeg,jpg,png,gif}')
+    ---
+    <div class="card">
+        <h2>{developer}</h2>
+        <p>Age: {age}</p>
+        <Image src={} alt={altText} />
+    </div>
+    ```
+
+6. Inside the `src` attribute, pass in the `images` object and use bracket notation for the image path. Then make sure to invoke the glob function. 
+   
+    ```astro title="src/components/MyCustomCardComponent.astro" ins={17}
+    ---
+     import { Image } from 'astro:assets';
+
+     interface Props {
+        imagePath: string;
+        altText: string;
+        name: string;
+        age: number;
+     }
+     
+     const { imagePath, altText, name, age } = Astro.props;
+     const images = import.meta.glob('/src/**/*.{jpeg,jpg,png,gif}')
+    ---
+    <div class="card">
+        <h2>{developer}</h2>
+        <p>Age: {age}</p>
+        <Image src={images[imagePath]()} alt={altText} />
+    </div>
+    ```
+
+:::note
+`images` is an object that contains all of the image paths inside the `assets` folder. 
+
+```
+const images = {
+  './assets/avatar-1.jpg': () => import('./assets/avatar-1.jpg'),
+  './assets/avatar-2.png': () => import('./assets/avatar-2.png'),
+  './assets/avatar-3.jpeg': () => import('./assets/avatar-3.jpeg')
+}
+```
+
+The `imagePath` prop is a string that contains the path to the image that you want to display. The `import.meta.glob()` is doing the work of finding the image path that matches the `imagePath` prop and handling the import for you.
+::: 
+
+7. Import and use the card component inside an Astro page, passing in the values for the `props`. 
+
+    ```astro title="src/pages/index.astro" 
+    ---
+    import MyCustomCardComponent from '../components/MyCustomCardComponent.astro';
+    ---
+    <MyCustomCardComponent 
+        imagePath="../assets/avatar-1.jpg"
```

Comment:
I couldn't get the relative path to work on my demo but absolute paths seem to work well.

```suggestion
        imagePath="/src/assets/avatar-1.jpg"
```

Reply:
Curious! I would expect relative paths to work, but this should work too!